components/ui/BannerCarousel.tsx

```diff
@@ -45,184 +53,124 @@ export interface Props {
-    desktop,
-    action,
-  } = image;
+function BannerItem({
+  image,
+  lcp,
+  id,
+}: {
+  image: Banner;
+  lcp?: boolean;
+  id: string;
+}) {
+  const { alt, mobile, desktop, link = "", action } = image;
 
   return (
     <a
       id={id}
-      href={action?.href ?? "#"}
-      aria-label={action?.label}
-      class="relative h-[600px] overflow-y-hidden w-full"
+      href={link}
+      aria-label={alt}
+      class="relative  overflow-y-hidden w-full"
     >
       <Picture preload={lcp}>
         <Source
           media="(max-width: 767px)"
           fetchPriority={lcp ? "high" : "auto"}
           src={mobile}
           width={360}
-          height={600}
+          height={441}
         />
         <Source
           media="(min-width: 768px)"
           fetchPriority={lcp ? "high" : "auto"}
           src={desktop}
           width={1440}
-          height={600}
+          height={441}
         />
         <img
-          class="object-cover w-full h-full"
+          class="object-cover w-full"
           loading={lcp ? "eager" : "lazy"}
           src={desktop}
           alt={alt}
         />
       </Picture>
-      {action && (
-        <div class="absolute h-min top-0 bottom-0 m-auto left-0 right-0 sm:right-auto sm:left-[12%] max-h-min max-w-[400px] flex flex-col gap-4 p-4 rounded glass">
-          <span class="text-6xl font-medium text-base-100">
-            {action.title}
-          </span>
-          <span class="font-medium text-xl text-base-100">
-            {action.subTitle}
-          </span>
-          <Button class="glass">{action.label}</Button>
-        </div>
-      )}
+
+      <BannerActionLayout action={action} />
     </a>
   );
 }
 
-function Dots({ images, interval = 0 }: Props) {
-  return (
-    <>
-      <style
-        dangerouslySetInnerHTML={{
-          __html: `
-          @property --dot-progress {
-            syntax: '<percentage>';
-            inherits: false;
-            initial-value: 0%;
-          }
-          `,
-        }}
-      />
-      <ul class="carousel justify-center col-span-full gap-4 z-10 row-start-4">
-        {images?.map((_, index) => (
-          <li class="carousel-item">
-            <Slider.Dot index={index}>
-              <div class="py-5">
-                <div
-                  class="w-16 sm:w-20 h-0.5 rounded group-disabled:animate-progress bg-gradient-to-r from-base-100 from-[length:var(--dot-progress)] to-[rgba(255,255,255,0.4)] to-[length:var(--dot-progress)]"
-                  style={{ animationDuration: `${interval}s` }}
-                />
-              </div>
-            </Slider.Dot>
-          </li>
-        ))}
-      </ul>
-    </>
-  );
-}
+export const BannerActionLayout = ({ action }: { action?: ActionProps }) => {
+  if (action) {
+    return (
+      <div class=" absolute lg:max-w-3xl p-4 lg:p-0  md:top-[-20px]  max-[900px]:top-[-20px]  h-min top-0 bottom-0 m-auto left-0 right-0 text-center  max-h-min  flex flex-col items-center">
```

Comment:
Voce adiciona muitos espaços desnecessarios nas classes CSS, tente evitar essa pratica para garantir que seu codigo seja o melhor possivel.
```suggestion
      <div class="absolute lg:max-w-3xl p-4 lg:p-0 md:top-[-20px] max-[900px]:top-[-20px] h-min top-0 bottom-0 m-auto left-0 right-0 text-center max-h-min flex flex-col items-center">
```

Reply:
Outra coisa, evita breakpoints diferentes do sm md lg xl 2xl, voce ta criando uma necessidade de max-[900px] que é so aplicada aqui. Acho que vale usar o `lg` nesse caso. Outro detalhe importante, prefira seguir com mobile first evitando o uso do prefixo `max-XX` e usando apenas os de min-width